packages/browser/src/browser/index.ts

```diff
@@ -169,9 +167,7 @@ async function flushFinalBuffer(
 ): Promise<void> {
   // Call popSnippetWindowBuffer before each flush task since there may be
   // analytics calls during async function calls.
-  buffer.push(...popSnippetWindowBuffer())
```

Comment:
Refactor: encapsulated this into the Buffer logic itself, so it occurs every time "buffer.getCalls()" is called